src/website/webview/views.py

```diff
@@ -1,16 +1,363 @@
+                        ),
+                        default=Value(0.0),
+                    ),
+                )
+                .values("id")
+                .annotate(
+                    max_rank_container=Max("rank_container"),
+                    max_rank_affected=Max("rank_affected"),
+                    max_rank_cpes=Max("rank_cpes"),
+                    max_rank_descriptions=Max("rank_descriptions"),
+                    title=Max("title"),
+                    cve_id=Max("cve__id"),
+                    cve_id_code=Max("cve__cve_id"),
+                    descriptions=ArrayAgg("descriptions__value"),
+                    affected_product=Max("affected__product"),
+                    affected_package_name=Max("affected__package_name"),
+                    affected_repo=Max("affected__repo"),
+                    affected_vendor=Max("affected__vendor"),
+                    affected_cpes=ArrayAgg("affected__cpes__name"),
+                )
+                .order_by(
+                    "-max_rank_container",
+                    "-max_rank_affected",
+                    "-max_rank_cpes",
+                    "-max_rank_descriptions",
+                    "id",
+                )
+            )
+
+            # To efficiently count is best to filter all relevant m2m related fields
+            # disregarding their order. Otherwise, the count will and paginator will
+            # force grouping (which is costly in this case) twice.
+            cve_count_function = self._cve_efficient_filter(
+                search_cves=search_cves
+            ).count
+
+            return cve_qs, cve_count_function
+        else:
+            cve_qs = (
+                Container.objects.values("id")
+                .annotate(
+                    title=Max("title"),
+                    cve_id=Max("cve__id"),
+                    cve_id_code=Max("cve__cve_id"),
+                    descriptions=ArrayAgg("descriptions__value"),
+                    affected_product=Max("affected__product"),
+                    affected_package_name=Max("affected__package_name"),
+                    affected_repo=Max("affected__repo"),
+                    affected_vendor=Max("affected__vendor"),
+                    affected_cpes=ArrayAgg("affected__cpes__name"),
+                )
+                .order_by("-cve__cve_id")
+            )
+
+            cve_count_function = Container.objects.values("id").count
+
+            return cve_qs, cve_count_function
+
+    def _get_pkg_qs(
+        self, search_pkgs: str | None
+    ) -> tuple["ValuesQuerySet[NixDerivation, dict[str, Any]]", Callable[[], int]]:
+        if search_pkgs:
+            # Do a 2-rank search to prevent description contents from penalizing hits on "name" and "attribute"
+            search_query = SearchQuery(search_pkgs)
+            # Check https://www.postgresql.org/docs/current/textsearch-controls.html#TEXTSEARCH-RANKING
+            # for the meaning of normalization values.
```

Comment:
Copy that comment to the other function, so a cursory reader will not miss it

Reply:
Applied